lib/Conversion/SCFToCalyx/SCFToCalyx.cpp

```diff
@@ -1002,36 +1042,36 @@ LogicalResult BuildOpGroups::buildOp(PatternRewriter &rewriter,
                                      MulIOp mul) const {
   Location loc = mul.getLoc();
   Type width = mul.getResult().getType(), one = rewriter.getI1Type();
-  auto multPipe =
+  auto mulPipe =
       getComponentState().getNewLibraryOpInstance<calyx::MultPipeLibOp>(
           rewriter, loc, {width, width, one, one, one, width, one});
-  // Pass the result from the MulIOp to the Calyx primitive.
-  mul.getResult().replaceAllUsesWith(multPipe.out());
-  auto reg = createReg(getComponentState(), rewriter, mul.getLoc(),
-                       getComponentState().getUniqueName("mult_reg"),
-                       width.getIntOrFloatBitWidth());
-  // Multiplication pipelines are not combinational, so a GroupOp is required.
-  auto group = createGroupForOp<calyx::GroupOp>(rewriter, mul);
-  getComponentState().addBlockScheduleable(mul->getBlock(), group);
-
-  rewriter.setInsertionPointToEnd(group.getBody());
-  rewriter.create<calyx::AssignOp>(loc, multPipe.left(), mul.getLhs());
-  rewriter.create<calyx::AssignOp>(loc, multPipe.right(), mul.getRhs());
-  // Write the output to this register.
-  rewriter.create<calyx::AssignOp>(loc, reg.in(), multPipe.out());
-  // The write enable port is high when the pipeline is done.
-  rewriter.create<calyx::AssignOp>(loc, reg.write_en(), multPipe.done());
-  rewriter.create<calyx::AssignOp>(
-      loc, multPipe.go(), getComponentState().getConstant(rewriter, loc, 1, 1));
-  // The group is done when the register write is complete.
-  rewriter.create<calyx::GroupDoneOp>(loc, reg.done());
+  auto output = mulPipe.out();
```

Comment:
Nit: Can we either give this a type, or just use `mulPipe.out()` below, since it is only used once? You could even do
```cpp
buildLibraryPipeOp<calyx::MultPipeLibOp>(rewriter, mul, mulPipe, /*out=*/mulPipe.out());
```